src/main/java/gameframework/game/GameUniverseDefaultImpl.java

```diff
@@ -21,6 +21,10 @@
 	@Override
 	public synchronized void addGameEntity(GameEntity gameEntity) {
 		gameEntities.add(gameEntity);
+		removeOverlappableAndBlockerEntity(gameEntity);
+	}
+
+	protected void removeOverlappableAndBlockerEntity(GameEntity gameEntity) {
```

Comment:
I don't understand why in `removeOverlappableAndBlockerEntity` you call `add*` methods